File: sdks/insuretech-go-sdk/pkg/models/business_beneficiary.go
package models

import (
	"time"
)

// BusinessBeneficiary represents a business_beneficiary
type BusinessBeneficiary struct {
	ActivePoliciesCount int `json:"active_policies_count,omitempty"`
	AuditInfo *AuditInfo `json:"audit_info,omitempty"`
	BeneficiaryId string `json:"beneficiary_id,omitempty"`
	BinNumber string `json:"bin_number,omitempty"`
	BusinessAddress *Address `json:"business_address,omitempty"`
	BusinessName string `json:"business_name,omitempty"`
	BusinessNameBn string `json:"business_name_bn,omitempty"`
	BusinessType *BusinessType `json:"business_type,omitempty"`
	ContactInfo *ContactInfo `json:"contact_info,omitempty"`
	EmployeeCount int `json:"employee_count,omitempty"`
	FocalPersonContact *ContactInfo `json:"focal_person_contact,omitempty"`
	FocalPersonDesignation string `json:"focal_person_designation,omitempty"`
	FocalPersonName string `json:"focal_person_name,omitempty"`
	FocalPersonNid string `json:"focal_person_nid,omitempty"`
	Id string `json:"id,omitempty"`
	IncorporationDate time.Time `json:"incorporation_date,omitempty"`
	IndustrySector string `json:"industry_sector,omitempty"`
	PendingActionsCount int `json:"pending_actions_count,omitempty"`
	PrimaryContact *PrimaryContact `json:"primary_contact,omitempty"`
	RegisteredAddress *Address `json:"registered_address,omitempty"`
	RegistrationNumber string `json:"registration_number,omitempty"`
	TaxId string `json:"tax_id,omitempty"`
	TinNumber string `json:"tin_number,omitempty"`
	TotalEmployeesCovered int `json:"total_employees_covered,omitempty"`
	TotalPremiumAmount *Money `json:"total_premium_amount,omitempty"`
	TradeLicenseExpiryDate time.Time `json:"trade_license_expiry_date,omitempty"`
	TradeLicenseIssueDate time.Time `json:"trade_license_issue_date,omitempty"`
	TradeLicenseNumber string `json:"trade_license_number,omitempty"`
}

File: backend/insurance_engine/src/InsuranceEngine.Beneficiary/Domain/Entities/BusinessBeneficiary.cs
using InsuranceEngine.Beneficiary.Domain.Enums;
using InsuranceEngine.SharedKernel.Domain;
using InsuranceEngine.SharedKernel.Domain.ValueObjects;

namespace InsuranceEngine.Beneficiary.Domain.Entities;

public class BusinessBeneficiary : Entity<Guid>
{
    public Guid BeneficiaryId { get; private set; }
    
    public string BusinessName { get; private set; } = string.Empty;
    public string? BusinessNameBn { get; set; }
    public string TradeLicenseNumber { get; private set; } = string.Empty;
    public string TinNumber { get; private set; } = string.Empty;
    
    public ContactInfo ContactInfo { get; set; } = new();
    public Address RegisteredAddress { get; set; } = new();
    public Address BusinessAddress { get; set; } = new();
    
    public BusinessType BusinessType { get; set; }
    public string? IndustrySector { get; set; }
    public int? EmployeeCount { get; set; }
    public DateTime? IncorporationDate { get; set; }
    
    public string? TradeLicenseIssueDate { get; set; }
    public string? TradeLicenseExpiryDate { get; set; }
    public string? BinNumber { get; set; }
    public string? RegistrationNumber { get; set; }
    public string? TaxId { get; set; }

    public string FocalPersonName { get; set; } = string.Empty;
    public ContactInfo FocalPersonContact { get; set; } = new();
    public string? FocalPersonDesignation { get; set; }
    public string? FocalPersonNid { get; set; }
    
    public int ActivePoliciesCount { get; set; }
    public int PendingActionsCount { get; set; }
    public int TotalEmployeesCovered { get; set; }
    
    public Money TotalPremiumAmount { get; set; } = Money.Zero;
    
    public AuditInfo AuditInfo { get; private set; } = new();

    public void UpdateFocalPerson(string name, string mobile)
    {
        FocalPersonName = name;
        FocalPersonContact = new ContactInfo(mobile);
        AuditInfo = AuditInfo with { UpdatedAt = DateTime.UtcNow };
    }

    // EF Core constructor
    public BusinessBeneficiary() { }

    public BusinessBeneficiary(
        Guid id,
        Guid beneficiaryId,
        string businessName,
        string tradeLicense,
        string tin) : base(id)
    {
        BeneficiaryId = beneficiaryId;
        BusinessName = businessName;
        TradeLicenseNumber = tradeLicense;
        TinNumber = tin;
        AuditInfo = new AuditInfo();
    }
}
